### PR TITLE
Compatible with 1.x openai

### PR DESCRIPTION
**Pull request by zhanghy-sketchzh**
### 🤔 What is the nature of this change? / 这个变动的性质是？

- [ ] New feature / 新特性提交
- [ ] Fix bug / bug 修复
- [x] Refactor code or style / 重构代码或样式
- [ ] Performance optimization / 性能优化
- [ ] Build optimization / 构建优化
- [ ] Website, documentation, demo improvements / 网站、文档、Demo 改进
- [ ] Test related / 测试相关
- [ ] Other / 其他

### 💡 Background or solution / 需求背景和解决方案

Compatible with 1.x openai

### 📝 Changelog / 更新日志

1. Judging and compatibility according to the user's OpenAI version
2.  Modify the incoming parameters to accommodate the new version of chat.completion
3. Parse the class format of the new version of completion response so that the output follows the dictionary format of the old version

### CODE REVIEW

File: XAgent/ai_functions/request/openai.py [hunk 1]
```diff
@@ -1,42 +1,140 @@
...
+
+    response = openai_function(**chatcompletion_kwargs)
+    return response
 
 
 def chatcompletion_request(**kwargs):
-    model_name = get_model_name(kwargs.pop('model',CONFIG.default_completion_kwargs['model']))
+    model_name = get_model_name(
+        kwargs.pop("model", CONFIG.default_completion_kwargs["model"])
+    )
```
Comment (Umpire2018) [inline XAgent/ai_functions/request/openai.py:83]: I like the style change here.

**Review by Umpire2018 — CHANGES_REQUESTED** [review]
1. This is a new feature instead of code refactor.
2. Please add more unit test if you are available and attach your test result here.
3. Please add more doc-string for this function.

Thanks for you contribution!

**Comment by semikolon** [comment]
What's blocking this? I don't understand.

**Comment by Umpire2018** [comment]
@semikolon 
Please kindly refer to [here](https://github.com/OpenBMB/XAgent/pull/189#pullrequestreview-1724155998). Do you suggest a fast merge of this PR? What is your reason? Our team has unanimously agreed that this PR needs to wait for thorough testing, add unit tests, and undergo a complete run-through to ensure no issues arise.

**Comment by zhanghy-sketchzh** [comment]
running_records:[2023_11_13_09_57_250800778c.zip](https://github.com/OpenBMB/XAgent/files/13333903/2023_11_13_09_57_250800778c.zip)

**Pull request closed without merging**